com/williamfiset/datastructures/binarysearchtree/SplayTree.java

```diff
@@ -0,0 +1,274 @@
+import com.williamfiset.datastructures.utils.TreePrinter;
+
+import java.util.ArrayList;
+import java.util.Scanner;
+
+
+class BinaryTree<T extends Comparable<T>> implements TreePrinter.PrintableNode {
+    private BinaryTree<T> left, right;
+    private T data;
+
+    public BinaryTree(T data) {
```

Comment:
We never want to allow null data into the tree. Please check this and throw an error. Same the setData method.

Reply:
Great Advice,Thanks !